scripts/prepare_heat_demand.py

```diff
@@ -0,0 +1,367 @@
+    demand["ghd" + "_" + carrier] = bdew.HeatBuilding(
+        demand.index,
+        holidays=holidays,
+        temperature=temperature,
+        shlp_type="ghd",
+        wind_class=0,
+        annual_heat_demand=yearly_demands["ghd" + "_" + carrier][0],
+        name="ghd",
+        ww_incl=True,
+    ).get_bdew_profile()
+
+    # Calculate total heat load in year
+    heat_load_year[carrier + "-load-profile"] = demand.sum(axis=1)
+    # TODO: Normalize heat load profile
+
+    return heat_load_year
+
+
+def postprocess_data(heat_load):
+    """
+    This function stacks time series of heat load profile and addes it to result DataFrame
+
+    Parameters
+    ----------
+    heat_load : pd.Dataframe
+        Empty Dataframe as result DataFrame
+
+    Returns
+    -------
+    heat_load : pd.DataFrame
+         DataFrame that contains the stacked heat load profile
+
+    """
+    # Stack time series with total heat load in year
+    heat_load_year_stacked = dp.stack_timeseries(heat_load_year)
+
+    heat_load_year_stacked["region"] = [region] * len(CARRIERS)
+    heat_load_year_stacked["scenario"] = [SCENARIO] * len(CARRIERS)
+    heat_load_year_stacked["var_unit"] = [sc_demand_unit] * len(CARRIERS)
+
+    # Append stacked heat load of year to stacked time series with total heat loads
+    heat_load = pd.concat([heat_load, heat_load_year_stacked], ignore_index=True)
+
+    return heat_load
+
+
+if __name__ == "__main__":
+    # path_this_file = os.path.realpath(__file__)
+    # raw_data = os.path.abspath(
+    #     os.path.join(path_this_file, os.pardir, os.pardir, "raw")
+    # )
+    # schema = os.path.abspath(
+    #     os.path.join(path_this_file, os.pardir, os.pardir, "oemof_b3", "schema")
+    # )
+    # results = os.path.abspath(
+    #     os.path.join(path_this_file, os.pardir, os.pardir, "results", "_resources")
+    # )
+    #
+    # in_path1 = os.path.join(raw_data, "weatherdata")  # path to weather data
+    # in_path2 = os.path.join(
+    #     raw_data, "distribution_households.csv"
+    # )  # path to household distributions data
+    # in_path3 = os.path.join(raw_data, "holidays.csv")  # path to holidays
+    # in_path4 = os.path.join(raw_data, "scalars.csv")  # path to b3 schema scalars.csv
+    # out_path = os.path.join(results, "load_profile_heat.csv")
+
+    in_path1 = sys.argv[1]  # path to weather data
+    in_path2 = sys.argv[2]  # path to household distributions data
+    in_path3 = sys.argv[3]  # path to holidays
+    in_path4 = sys.argv[4]  # path to b3 scalars.csv
+    out_path = sys.argv[5]
+
+    REGION = ["BB", "BE"]
+    SCENARIO = "base"
+    CARRIERS = ["heat_central", "heat_decentral"]
+
+    # Read time series template
+    ts_header = dp.HEADER_B3_TS
+
+    # Add empty data frame for results / output
+    heat_load = pd.DataFrame(columns=ts_header)
+
+    for region in REGION:
+
+        weather_data = prepare_weather_data()
+        years = get_years()
+
+        for index, year in enumerate(years):
```

Comment:
Is the index of the years always the same as the index of the weather data?

Reply:
Yes they should be. `years` is obtained from file name of weather data. 
I added some exceptions with commit https://github.com/rl-institut/oemof-B3/pull/76/commits/187bf0176f785fe7f14c78a2025e6d230e894749 which should guarantee that the index is the same.

Does that resolve your concerns [here](https://github.com/rl-institut/oemof-B3/pull/76#discussion_r755077358)?

Reply:
Resolved as described in https://github.com/rl-institut/oemof-B3/pull/76#discussion_r756316977.